deepdrivemd/deepdrivemd_stream.py

```diff
@@ -0,0 +1,253 @@
+            cfg.task_config.node_local_path = self.cfg.node_local_path
+            cfg.task_config.output_path = output_path
+
+            # Write yaml configuration
+            cfg_path = stage_api.config_path(self.stage_idx, task_idx)
+            cfg.task_config.dump_yaml(cfg_path)
+            task = generate_task(cfg)
+            task.arguments += ["-c", cfg_path.as_posix()]
+            stage.add_tasks(task)
+
+        return stage
+
+    def generate_machine_learning_stage(self) -> Stage:
+        stage = Stage()
+        stage.name = self.MACHINE_LEARNING_STAGE_NAME
+        cfg = self.cfg.machine_learning_stage
+        stage_api = self.api.machine_learning_stage
+
+        task_idx = 0
+        output_path = stage_api.task_dir(self.stage_idx, task_idx, mkdir=True)
+        assert output_path is not None
+
+        # Update base parameters
+        cfg.task_config.experiment_directory = self.cfg.experiment_directory
+        cfg.task_config.stage_idx = self.stage_idx
+        cfg.task_config.task_idx = task_idx
+        cfg.task_config.node_local_path = self.cfg.node_local_path
+        cfg.task_config.output_path = output_path
+        cfg.task_config.model_tag = stage_api.unique_name(output_path)
+        if self.stage_idx > 0:
+            # Machine learning should use model selection API
+            cfg.task_config.init_weights_path = None
+
+        # Write yaml configuration
+        cfg_path = stage_api.config_path(self.stage_idx, task_idx)
+        cfg.task_config.dump_yaml(cfg_path)
+        task = generate_task(cfg)
+        task.arguments += ["-c", cfg_path.as_posix()]
+        stage.add_tasks(task)
+
+        return stage
+
+    def generate_agent_stage(self) -> Stage:
+        stage = Stage()
+        stage.name = self.AGENT_STAGE_NAME
+        cfg = self.cfg.agent_stage
+        stage_api = self.api.agent_stage
+
+        task_idx = 0
+        output_path = stage_api.task_dir(self.stage_idx, task_idx, mkdir=True)
+        assert output_path is not None
+
+        # Update base parameters
+        cfg.task_config.experiment_directory = self.cfg.experiment_directory
+        cfg.task_config.stage_idx = self.stage_idx
+        cfg.task_config.task_idx = task_idx
+        cfg.task_config.node_local_path = self.cfg.node_local_path
+        cfg.task_config.output_path = output_path
+
+        # Write yaml configuration
+        cfg_path = stage_api.config_path(self.stage_idx, task_idx)
+        cfg.task_config.dump_yaml(cfg_path)
+        task = generate_task(cfg)
+        task.arguments += ["-c", cfg_path.as_posix()]
+        stage.add_tasks(task)
+
+        return stage
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+    cfg = StreamingExperimentConfig.from_yaml(args.config)
+    cfg.config_directory = os.path.dirname(os.path.abspath(args.config))
+    print("config_directory = ", cfg.config_directory)
+
+    cfg.adios_xml_sim = f"{cfg.config_directory}/adios_sim.xml"
```

Comment:
Both `cfg.adios_xml_sim` and `cfg.adios_xml_agg` are type Path. We should avoid having types change at runtime. The fix is:

```
cfg.adios_xml_sim = Path(cfg.config_directory) / "adios_sim.xml"
cfg.adios_xml_agg = Path(cfg.config_directory) / "adios_agg.xml"
``` 

Or better yet I think this would work:
```
cfg.config_directory = args.config.resolve().parent
cfg.adios_xml_sim = cfg.config_directory / "adios_sim.xml"
cfg.adios_xml_agg = cfg.config_directory / "adios_agg.xml"

```

Reply:
Done